src/flypad-backend/src/Aircraft/AircraftProcedures.h

```diff
@@ -29,272 +28,264 @@ struct ProcedureStep {
-  new ProcedureStep {"WX Radar Off",      4100, false, 1000, "(L:XMLVAR_A320_WEATHERRADAR_SYS) 1 ==",                       "1 (>L:XMLVAR_A320_WEATHERRADAR_SYS)"},
+static const std::vector<ProcedureStep> TAKEOFF_CONFIG_OFF = {
+  {"LL Lt L Off",       4060, false, 0,    "(A:CIRCUIT SWITCH ON:18, Bool) ! (L:LANDING_2_RETRACTED) &&", "2 (>L:LIGHTING_LANDING_2) 1 (>L:LANDING_2_RETRACTED) (A:CIRCUIT SWITCH ON:18, Bool) if{ 18 (>K:ELECTRICAL_CIRCUIT_TOGGLE)"},
+  {"LL Lt R Off",       4070, false, 1000, "(A:CIRCUIT SWITCH ON:19, Bool) ! (L:LANDING_3_RETRACTED) &&", "2 (>L:LIGHTING_LANDING_3) 1 (>L:LANDING_3_RETRACTED) (A:CIRCUIT SWITCH ON:19, Bool) if{ 19 (>K:ELECTRICAL_CIRCUIT_TOGGLE)"},
+  {"NOSE Lt Takeoff",   4080, false, 2000, "(A:CIRCUIT SWITCH ON:17, Bool) !",                            "(A:CIRCUIT SWITCH ON:17, Bool) if{ 17 (>K:ELECTRICAL_CIRCUIT_TOGGLE)"},
+  {"Strobe Auto",       2180, false, 1000, "(A:LIGHT STROBE, Bool) !",                                    "1 (>L:STROBE_0_AUTO) 0 (>K:STROBES_OFF)"},
+  {"TCAS Switch TA/RA", 4090, false, 1000, "(L:A32NX_SWITCH_TCAS_POSITION) 0 ==",                         "0 (>L:A32NX_SWITCH_TCAS_POSITION)"},
+  {"WX Radar Mode",     4110, false, 1000, "(L:XMLVAR_A320_WEATHERRADAR_MODE) 1 ==",                      "1 (>L:XMLVAR_A320_WEATHERRADAR_MODE)"},
+  {"WX Radar Off",      4100, false, 1000, "(L:XMLVAR_A320_WEATHERRADAR_SYS) 1 ==",                       "1 (>L:XMLVAR_A320_WEATHERRADAR_SYS)"},
 };
 // @formatter:on
 
 class AircraftProcedures {
-  vector<ProcedureStep*>* coldAndDark = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* powered = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* readyForPushback = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* readyForTaxi = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* readyForTakeoff = new vector<ProcedureStep*>;
-  // just to get a list of IDs
-  vector<ProcedureStep*>* all = new vector<ProcedureStep*>;
+  std::vector<const ProcedureStep*> coldAndDark;
+  std::vector<const ProcedureStep*> powered;
+  std::vector<const ProcedureStep*> readyForPushback;
+  std::vector<const ProcedureStep*> readyForTaxi;
+  std::vector<const ProcedureStep*> readyForTakeoff;
+
+  void printProcedure(const std::vector<const ProcedureStep*>& procedures) {
+    std::for_each(begin(procedures), end(procedures), [](const auto* procedure) {
+        std::cout << procedure->id << " = " << procedure->description << std::endl;
+    });
+  }
+
+  void doInsertProcedures(std::vector<const ProcedureStep*>& dest, const std::vector<ProcedureStep>& src) {
+    std::transform(begin(src), end(src), back_inserter(dest), [](const auto& procedure) {
+      return &procedure;
+    });
+  }
+
+    void insertProcedures(std::vector<const ProcedureStep*>& dest, const std::vector<ProcedureStep>& a, const std::vector<ProcedureStep>& b,
```

Comment:
Same here: Why not static member function? And also this function would be more general if it would accept an unspecified number of arguments instead of a fixed number of vectors (lookup variadic functions or you can use std::initializer_list because all have the same type).

Reply:
Initialiazer list would be possible, however I think it's cleaner to use variadic template arguments. I've added a new push, let me know what you think.